src/storage/store/string_column_chunk.cpp

```diff
@@ -128,12 +136,72 @@ void StringColumnChunk::setValueFromString(const char* value, uint64_t length, u
     offsetChunk->setValue<string_offset_t>(startOffset, index);
     offsetChunk->setNumValues(index + 1);
     ColumnChunk::setValue<string_index_t>(index, pos);
+    indexTable.insert(std::make_pair(std::string(value, length), index));
+}
+
+void StringColumnChunk::prepareForFlush() {
+    // Only de-duplicate strings, not blobs
+    if (dataType.getLogicalTypeID() != LogicalTypeID::STRING) {
+        return;
+    }
+    // Prune unused entries in the dictionary before we flush
+    // We already de-duplicate as we go, but when out of place updates occur new values will be
```

Comment:
I guess we should only need to do this when there are updates to the column chunk, and `append` should just be fine and this function is unnecessary for that?